zerver/tests/test_bugdown.py

```diff
@@ -892,6 +898,210 @@ def render(msg: Message, content: str) -> str:
+
+        def render(msg: Message, content: str) -> str:
+            return render_markdown(msg,
+                                   content,
+                                   realm_alert_words_automaton=realm_alert_words_automaton,
+                                   user_ids = user_ids)
+
+        content = "hello how is this possible how are you doing today"
+        render(msg, content)
+        expected_user_ids = {user_profiles['hamlet'].id, user_profiles['cordelia'].id, user_profiles['iago'].id, user_profiles['prospero'].id, user_profiles['othello'].id}  # type: Set[int]
+        self.assertEqual(msg.user_ids_with_alert_words, expected_user_ids)
+
+    def test_alert_words_returns_user_ids_with_alert_words_1(self) -> None:
+        alert_words_for_users = {
+            'hamlet': ['provisioning', 'Prod deployment'],
+            'cordelia': ['test', 'Prod'],
+            'iago': ['prod'], 'prospero': ['deployment'],
+            'othello': ['last']
+        }  # type: Dict[str, List[str]]
+        user_profiles = {}  # type: Dict[str, UserProfile]
+        user_ids = set()  # type: Set[int]
+        for (username, alert_words) in alert_words_for_users.items():
+            user_profile = self.example_user(username)
+            user_profiles.update({username: user_profile})
+            user_ids.add(user_profile.id)
+            do_set_alert_words(user_profile, alert_words)
+        sender_user_profile = self.example_user('polonius')
+        user_ids.add(sender_user_profile.id)
+        msg = Message(sender=sender_user_profile, sending_client=get_client("test"))
+        realm_alert_words_automaton = get_alert_word_automaton(sender_user_profile.realm)
+
+        def render(msg: Message, content: str) -> str:
+            return render_markdown(msg,
+                                   content,
+                                   realm_alert_words_automaton=realm_alert_words_automaton,
+                                   user_ids = user_ids)
+
+        content = """Hello, everyone. Prod deployment has been completed
+        And this is a new line
+        to test out how markdown convert this into something line ending splitted array
+        and this is a new line
+        last"""
+        render(msg, content)
+        expected_user_ids = {
+            user_profiles['hamlet'].id,
+            user_profiles['cordelia'].id,
+            user_profiles['iago'].id,
+            user_profiles['prospero'].id,
+            user_profiles['othello'].id
+        }  # type: Set[int]
+        self.assertEqual(msg.user_ids_with_alert_words, expected_user_ids)
+
+    def test_alert_words_returns_user_ids_with_alert_words_in_french(self) -> None:
+        alert_words_for_users = {
+            'hamlet': ['réglementaire', 'une politique', 'une merveille'],
+            'cordelia': ['énormément', 'Prod'],
+            'iago': ['prod'], 'prospero': ['deployment'],
```

Comment:
Can you add a comment here that iago, cordelia, and othello have strings that don't appear in the text and thus should not match?  It'd make the test a bit more readable.